### PR TITLE
Setup workflows component library

### PR DESCRIPTION
**Pull request by iamvigneshwars**

### CODE REVIEW

File: UI/.devcontainer/Dockerfile
Comment (garryod) [inline UI/.devcontainer/Dockerfile:1]: Could we just reference this upstream container in our `devcontainer.json`?

File: UI/.devcontainer/docker-compose.yaml
Comment (garryod) [inline UI/.devcontainer/docker-compose.yaml:1]: Do we need to be using `docker-compose` here if it's just one service?

File: UI/workflows-components/.gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+pnpm-debug.log*
+lerna-debug.log*
+
```
Comment (garryod) [inline UI/workflows-components/.gitignore:9]: This middle block is missing a comment

File: UI/workflows-components/vite.config.ts
Comment (garryod) [inline UI/workflows-components/vite.config.ts:1]: Can we drop the vite stuff from the component library?

File: UI/workflows-components/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "dev": "vite",
+    "build": "tsc -b && vite build",
+    "lint": "eslint .",
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@emotion/react": "^11.13.0",
+    "@emotion/styled": "^11.13.0",
+    "@mui/material": "^5.16.7",
+    "eslint-plugin-react": "^7.35.0",
```
Comment (garryod) [inline UI/workflows-components/package.json:16]: Can this be in `devDependencies`?

**Review by garryod — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -8,3 +8,5 @@ config_argus
 kubeconfig.yaml
 # Rust
 sessionspaces/target
+# node modules
+node_modules
```
Comment (garryod) [inline .gitignore:12]: Should this be at the repo root or alongside the `package.json`?

File: UI/workflows-lib/.gitignore
Comment (garryod) [inline UI/workflows-lib/.gitignore:1]: I think a lot of these want to be at the workspace level

File: UI/workflows-lib/.gitignore [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# Editor directories and files
+.vscode/*
+!.vscode/extensions.json
+.idea
+.DS_Store
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
```
Comment (garryod) [inline UI/workflows-lib/.gitignore:24]: Can probably remove these

**Review by garryod — COMMENTED** [review]
Looks pretty good to me, but have tagged @abigailalexander as someone who knows this stuff

**Comment by abigailalexander** [comment]
I think I might be missing something, but what is being built into a component library here? I can see references to `lib/main.ts` that doesn't seem to exist, nor any other `.ts` or `.tsx` files here. Is this PR just for setting up the config needed to build the future component library?

**Comment by iamvigneshwars** [comment]
This PR is just a setup to the component library. The actual components will be added in separate PR's. @abigailalexander

**Comment by abigailalexander** [comment]
Makes sense, thanks for clarifying! Looks good to me @iamvigneshwars

**Review by abigailalexander — DISMISSED** [review]

**Review by garryod — APPROVED** [review]
LGTM

**Review by garryod — CHANGES_REQUESTED** [review]
This is such a nit pick, but can we rename the `UI` directory to `frontend`?

**Review by garryod — APPROVED** [review]

**Pull request merged**